src/sc2laddercore/AgentsConfig.cpp

```diff
@@ -12,242 +12,242 @@
 #include <string>
 
 AgentsConfig::AgentsConfig(LadderConfig *InLadderConfig)
-	: PlayerIds(nullptr),
-	EnablePlayerIds(false),
-	Config(InLadderConfig)
+    : PlayerIds(nullptr),
+    EnablePlayerIds(false),
+    Config(InLadderConfig)
 
 {
-	if (Config == nullptr)
-	{
-		return;
-	}
-	std::string PlayerIdFile = Config->GetValue("PlayerIdFile");
-	if (PlayerIdFile.length() > 0)
-	{
-		PlayerIds = new LadderConfig(PlayerIdFile);
-		EnablePlayerIds = true;
-	}
-	if (Config->GetValue("BotConfigFile") != "")
-	{
-		LoadAgents("", Config->GetValue("BotConfigFile"));
-	}
-	else if (Config->GetValue("BaseBotDirectory") != "")
-	{
-		ReadBotDirectories(Config->GetValue("BaseBotDirectory"));
-	}
+    if (Config == nullptr)
+    {
+        return;
+    }
+    std::string PlayerIdFile = Config->GetStringValue("PlayerIdFile");
+    if (PlayerIdFile.length() > 0)
+    {
+        PlayerIds = new LadderConfig(PlayerIdFile);
+        EnablePlayerIds = true;
+    }
+    if (Config->GetStringValue("BotConfigFile") != "")
+    {
+        LoadAgents("", Config->GetStringValue("BotConfigFile"));
+    }
+    else if (Config->GetStringValue("BaseBotDirectory") != "")
+    {
+        ReadBotDirectories(Config->GetStringValue("BaseBotDirectory"));
+    }
 }
 
 void AgentsConfig::ReadBotDirectories(const std::string &BaseDirectory)
 {
-	BotConfigs.clear();
+    BotConfigs.clear();
     std::vector<std::string> directories;
     sc2::scan_directory(BaseDirectory.c_str(), directories, true, true);
-	for (const std::string  &Directory : directories)
-	{
+    for (const std::string  &Directory : directories)
+    {
         std::string CurrentConfigFile = Directory + "/ladderbots.json";
-		LoadAgents(Directory, CurrentConfigFile);
-	}
+        LoadAgents(Directory, CurrentConfigFile);
+    }
 
 }
 
 void AgentsConfig::LoadAgents(const std::string &BaseDirectory, const std::string &BotConfigFile)
 {
-	//	std::string BotConfigFile = Config->GetValue("BotConfigFile");
```

Comment:
Are you sure that indent should be changed? (here and in other places).

Reply:
That one not. And it is caused by not having a common indention over the different files.
It is super annoying/frustrating. That is why in the one file I made Ctrl+x Ctrl+v so that my IDE changed all lines to four spaces. I don't want to change the config in a per file basis.

Reply:
Well ok, but it should be mentioned somehow, e.g. in a separate commit or you shouldn't (ideally) change indent everywhere. (e.g. because it hides the latest history from developers).

So far PR states that a new option was added and do many-many unrelated changes. To be honest, hard to review.